src/app/hooks/useWalletReducer.ts

```diff
@@ -102,34 +105,43 @@ const useWalletReducer = () => {
     dispatch(getActiveAccountsAction(walletAccounts));
   };
 
-  const unlockWallet = async (password: string) => {
+  const migrateLegacySeedStorage = async (password: string) => {
     const pHash = await generatePasswordHash(password);
-    const decrypted = await decryptSeedPhrase(encryptedSeed, password);
+    await decryptSeedPhraseCBC(encryptedSeed, pHash.hash).then(async (decrypted) => {
+      await initSeedVault(password);
+      await storeSeed(decrypted);
+      localStorage.removeItem('salt');
+      dispatch(storeEncryptedSeedAction(''));
+      return decrypted;
+    });
+  };
+
+  const unlockWallet = async (password: string) => {
+    if (encryptedSeed && encryptedSeed.length > 0) {
+      await migrateLegacySeedStorage(password);
+      return;
+    }
+    const decrypted = await unlockVault(password);
```

Comment:
Won't this result in the unlock flow not completing on migration? Should it not continue after the migration and not return yet since it doesn't perform the functions after line 124?

```suggestion
    let decrypted: string
    if (encryptedSeed && encryptedSeed.length > 0) {
      decrypted = await migrateLegacySeedStorage(password);
    } else {
      decrypted = await unlockVault(password);
    }

// ... continue with funciton
```

Reply:
it was decided to not load the accounts when undergoing the initial migration